portable/release.sh

```diff
@@ -40,23 +37,33 @@ die "Usage: $0 [--output=<directory>] <version> [optional components]"
 test -d "$output_directory" ||
 die "Directory inaccessible: '$output_directory'"
 
-ARCH="$(uname -m)"
-case "$ARCH" in
-i686)
+case "$MSYSTEM" in
+MINGW32)
 	BITNESS=32
+	ARCH=i686
+	ARTIFACT_SUFFIX=32
 	MD_ARG=128M
 	;;
-x86_64)
+MINGW64)
+	BITNESS=64
+	ARCH=x86_64
+	ARTIFACT_SUFFIX=64
+	MD_ARG=256M
+	;;
+CLANGARM64)
 	BITNESS=64
+	ARCH=aarch64
+	ARTIFACT_SUFFIX=ARM64
 	MD_ARG=256M
 	;;
 *)
-	die "Unhandled architecture: $ARCH"
+	die "Unhandled MSYSTEM: $MSYSTEM"
 	;;
 esac
+MSYSTEM_LOWER=${MSYSTEM,,}
```

Comment:
Added the environment variable in https://github.com/git-for-windows/build-extra/pull/437/commits/17a3a4a6d54979c669289517d49b3c1ee70c4359

Reply:
The `MSYSTEM` variable is defined either in `/etc/profile` or in `git-bash.exe`, depending on the exact scenario, or in the case of CI runs [by the `setup-git-for-windows-sdk` Action](https://github.com/dscho/setup-git-for-windows-sdk/blob/0b1f2cf212dffda75a306481d91866af92278ac7/main.ts#L76). It should not be necessary to set it yourself unless it is set incorrectly.

Reply:
Here's an example of when it failed without me explicitly setting the `MSYSTEM` in CI: https://github.com/git-for-windows/build-extra/actions/runs/3357168162/jobs/5562796433. The `build-artifacts` CI job doesn't use `setup-git-for-windows-sdk` so that explains why it failed. If you want, I can set `MSYSTEM: mingw64` only to the `build-artifacts` CI job.

Reply:
No, setting `MSYSTEM` should be scoped to the particular SDK instance. In other words, [here](https://github.com/git-for-windows/build-extra/blob/fe12b23f586d34cad55c855f7edce0fcbb40c37f/.github/workflows/main.yml#L113), [here](https://github.com/git-for-windows/build-extra/blob/fe12b23f586d34cad55c855f7edce0fcbb40c37f/.github/workflows/main.yml#L167) and [here](https://github.com/git-for-windows/build-extra/blob/fe12b23f586d34cad55c855f7edce0fcbb40c37f/.github/workflows/main.yml#L203), we need to insert a line like this (in a separate, preparatory commit):

```sh
          echo "MSYSTEM=MINGW${{matrix.bitness}}" >>$GITHUB_ENV &&
```

(The first instance is not inside a matrix, it needs to hard-code `MINGW64`.)